### ISSUE TITLE
Cardiologist-Level Arrhythmia Detection with Convolutional Neural Networks

### ISSUE THREAD

**Issue by akundaje** [greenelab/deep-review#568]
Cardiologist-Level Arrhythmia Detection with Convolutional Neural Networks

https://arxiv.org/abs/1707.01836

We develop an algorithm which exceeds the performance of board certified cardiologists in detecting a wide range of heart arrhythmias from electrocardiograms recorded with a single-lead wearable monitor. We build a dataset with more than 500 times the number of unique patients than previously studied corpora. On this dataset, we train a 34-layer convolutional neural network which maps a sequence of ECG samples to a sequence of rhythm classes. Committees of board-certified cardiologists annotate a gold standard test set on which we compare the performance of our model to that of 6 other individual cardiologists. We exceed the average cardiologist performance in both recall (sensitivity) and precision (positive predictive value).

**Comment by 123helloworld123** [comment]
Hello, where can I find this paper's code? Thank you so much!

**Comment by agitter** [comment]
@123helloworld123 this repository focuses on a review manuscript and discussion of deep learning in biomedicine. None of us are affiliated with the manuscript linked above, so you'll have to try contacting the paper's authors.

If you do find the source code is available, we would be interested in having you share the link here as well.  Thanks.

**Comment by AiHrt** [comment]
Hi, anyone have the codes?

**Comment by AiHrt** [comment]
Hi, anyone's got the codes? could anyone please be so kind as to share it here?

**Comment by AiHrt** [comment]
I have a question regarding the paper, that is, what's the output shape of the proposed model?is it (None,14,30) where 14 refers to the class numbers and 30 refers to 30 seconds. Or (None,30) ? Because the paper said that the model output a prediction once every second, I am confused about this.

**Comment by AiHrt** [comment]
Another question is that how's the label arranged? Given a sample of 30 seconds(9000 points), is the corresponding label (1,1,1,1,1,2,2,2,2,2,2,2,2,2,2,2,2)(30 integers indicating class 1 2...)?

**Comment by FoadJafari** [comment]
Hi,

Can you tell me something about the test time on the wearable device (e.g. Moto 360 sport OR Moto 360 2nd gen)?

**Comment by agitter** [comment]
@AiHrt @FoadJafari please see my comment above:

> this repository focuses on a review manuscript and discussion of deep learning in biomedicine. None of us are affiliated with the manuscript linked above, so you'll have to try contacting the paper's authors.

**Comment by jeffreyzfzeng** [comment]
Hi,  Just wondering if you have some sample code available for others to take a look? Thanks.

**Comment by stephenra** [comment]
@jeffreyzfzeng Please read the earlier part of this thread. This is not a dedicated repository for the paper or associated code itself. A number of users have already asked about the code and this is the response from @agitter above:

> this repository focuses on a review manuscript and discussion of deep learning in biomedicine. None of us are affiliated with the manuscript linked above, so you'll have to try contacting the paper's authors.

**Comment by evancofer** [comment]
This has been published:

> Computerized electrocardiogram (ECG) interpretation plays a critical role in the clinical ECG workflow1. Widely available digital ECG data and the algorithmic paradigm of deep learning2 present an opportunity to substantially improve the accuracy and scalability of automated ECG analysis. However, a comprehensive evaluation of an end-to-end deep learning approach for ECG analysis across a wide variety of diagnostic classes has not been previously reported. Here, we develop a deep neural network (DNN) to classify 12 rhythm classes using 91,232 single-lead ECGs from 53,549 patients who used a single-lead ambulatory ECG monitoring device. When validated against an independent test dataset annotated by a consensus committee of board-certified practicing cardiologists, the DNN achieved an average area under the receiver operating characteristic curve (ROC) of 0.97. The average F1 score, which is the harmonic mean of the positive predictive value and sensitivity, for the DNN (0.837) exceeded that of average cardiologists (0.780). With specificity fixed at the average specificity achieved by cardiologists, the sensitivity of the DNN exceeded the average cardiologist sensitivity for all rhythm classes. These findings demonstrate that an end-to-end deep learning approach can classify a broad range of distinct arrhythmias from single-lead ECGs with high diagnostic performance similar to that of cardiologists. If confirmed in clinical settings, this approach could reduce the rate of misdiagnosed computerized ECG interpretations and improve the efficiency of expert human ECG interpretation by accurately triaging or prioritizing the most urgent conditions.

https://doi.org/10.1038/s41591-018-0268-3
